v5/core/client/custom_saas_environmet.go

```diff
@@ -0,0 +1,66 @@
+package client
+
+import (
+	"strings"
+
+	"github.com/rollout/rox-go/v5/core/model"
+)
+
+type CustomSaasEnvironment struct{
+	getConfigApiURL string
+	getConfigCloudURL string
+	sendStateApiURL string
+	sendStateCloudURL string
+	analyticsURL string
+	pushNotificationURL string
+}
+
+func TrimSuffix(s, suffix string) string {
+    if strings.HasSuffix(s, suffix) {
+        s = s[:len(s)-len(suffix)]
+    }
+    return s
+}
+
+func NewCustomSaasEnvironment(options model.NetworkConfigurationsOptions) CustomSaasEnvironment {
```

Comment:
Personally, I would just call this NewCustomEnvironment. It's all SaaS so it doesn't need repeating here.

Reply:
updated